mongo/options/clientoptions.go

```diff
@@ -110,6 +110,8 @@ type Credential struct {
 	Username                string
 	Password                string
 	PasswordSet             bool
+	OIDCMachineCallback     driver.OIDCCallback
+	OIDCHumanCallback       driver.OIDCCallback
```

Comment:
We need to canonically define or re-define the `OIDCCallback` type and associated structs here so the public API doesn't include symbols from the experimental API.

E.g. types that need to be defined in the `options` package:
```go
// OIDCCallback is the type for both Human and Machine Callback flows. RefreshToken will always be
// nil in the OIDCArgs for the Machine flow.
type OIDCCallback func(context.Context, *OIDCArgs) (*OIDCCredential, error)

// OIDCArgs contains the arguments for the OIDC callback.
type OIDCArgs struct {
	Version      int
	IDPInfo      *IDPInfo
	RefreshToken *string
}

// OIDCCredential contains the access token and refresh token.
type OIDCCredential struct {
	AccessToken  string
	ExpiresAt    *time.Time
	RefreshToken *string
}
```
It may be possible to canonically define the `OIDCCallback`, `OIDCArgs`, and `OIDCCredential` types in the `options` package, and use those types in the `driver` and auth` packages. If not, we will have to add logic to convert between the types.

Reply:
Tell me if this solution works for you, I think it's good.